Genode bindings failures on Debian Testing/Unstable

For some reasons `solo5-bindings-genode` fails on Debian Testing and Unstable:
```
# ld -nostdlib -z max-page-size=0x1000 -shared -gc-sections --eh-frame-hdr -Ttext=0x01000000 --dynamic-linker=ld.lib.so -rpath-link=. -T /home/opam/.opam/4.10/.opam-switch/build/solo5-bindings-genode.0.6.4/bindings/genode/genode_dyn.ld /home/opam/.opam/4.10/.opam-switch/build/solo5-bindings-genode.0.6.4/bindings/genode/solo5.lib.so \
#     test_tls.genode.o manifest.genode.o -o test_tls.genode
# ld: test_tls.genode: error: PHDR segment not covered by LOAD segment
# make[2]: *** [../Makefile.tests:73: test_tls.genode] Error 1
# make[2]: Leaving directory '/home/opam/.opam/4.10/.opam-switch/build/solo5-bindings-genode.0.6.4/tests/test_tls'
# make[1]: *** [GNUmakefile:33: test_tls] Error 2
# make[1]: Leaving directory '/home/opam/.opam/4.10/.opam-switch/build/solo5-bindings-genode.0.6.4/tests'
# make: *** [GNUmakefile:70: tests] Error 2
```
But I've seen this error before and it is unrelated to the package. Merging. Thanks a lot!

_Originally posted by @kit-ty-kate in https://github.com/ocaml/opam-repository/pull/15935#issuecomment-593514952_

This is presumably due to toolchain changes in Debian Testing/Unstable.

--- a/build.sh
+++ b/build.sh
@@ -46,6 +46,11 @@ do_basic()
 {
     message "Starting build."
     try ./configure.sh
+    # Genode bindings are disabled by default; enable them for Linux hosts to
+    # ensure they get tested by Solo5 CI.
+    if [ "$(uname -s)" = "Linux" ]; then
+        echo CONFIG_GENODE=1 >>Makeconf
+    fi
     try ${MAKE}
     # Some CIs can now run tests, so do that.
     if [ -n "${SURF_RUN_TESTS}" ]; then

--- a/configure.sh
+++ b/configure.sh
@@ -153,8 +153,8 @@ config_host_linux()
     fi
     [ "${CONFIG_ARCH}" = "x86_64" ] && CONFIG_VIRTIO=1
     [ "${CONFIG_ARCH}" = "x86_64" ] && CONFIG_MUEN=1
-    [ "${CONFIG_ARCH}" = "x86_64" ] && CONFIG_GENODE=1
     [ "${CONFIG_ARCH}" = "ppc64le" ] && CONFIG_HVT=
+    CONFIG_GENODE=
 }
 
 config_host_freebsd()

--- a/docs/building.md
+++ b/docs/building.md
@@ -45,7 +45,9 @@ to disable some _targets_, you can either use:
 make CONFIG_HVT= # disables hvt
 ```
 
-or edit the generated `Makeconf` before running `make`.
+or edit the generated `Makeconf` before running `make`. Conversely, using
+`CONFIG_TARGET=1` will enable a _target_ that was not explicitly enabled by
+configure.sh (for example, Genode).
 
 Note that the Solo5 Makefiles do not support building arbitrary `make` targets
 (e.g. `bindings` or `tests`).
@@ -85,7 +87,9 @@ Experimental:
   and executed as a native Genode component. The bindings library is provided
   here in the form of a stub library that can be built using a stock C
   toolchain along with the C++ implementation that must be built using the
-  Genode toolchain.
+  Genode toolchain. Note that the Genode _target_ must be enabled explicitly
+  using `CONFIG_GENODE=1`, and may not build with standard toolchains. If in
+  doubt, please consult with Genode developers.
 
 Limited:
 

